frontend/src/tests/UsernameInput.spec.ts

```diff
@@ -0,0 +1,171 @@
+import UserNameInput from '@/components/grantaccess/form/UserNameInput.vue';
+import { isLoading, setLoadingState } from '@/store/LoadingState';
+import type { VueWrapper } from '@vue/test-utils/dist/vueWrapper';
+import type { DOMWrapper } from '@vue/test-utils/dist/domWrapper';
+import type { AxiosResponse } from 'axios';
+import { fixPrimevueCssError } from '@/tests/common/fixPrimevueCssErr';
+
+fixPrimevueCssError()
+
+
+const userInputMock = (): AxiosResponse => {
+    return {
+        data: {
+            firstName: "Name",
+            found: true,
+            lastName: "LastName",
+            userId: 'userId'
+        },
+        status: 200,
+        statusText: 'Ok',
+        headers: {},
+        config: {},
+    }
+
+}
+
+describe('UserNameInput', () => {
+    let wrapper: VueWrapper;
+    let emitChange: unknown[][] | undefined;
+    let emitSetVerifyResult: unknown[][] | undefined;
+
+    let usernameInputText: DOMWrapper<HTMLElement>;
+    let usernameInputTextEl: HTMLInputElement;
+    let verifyButton: DOMWrapper<HTMLElement>;
+    let verifyButtonEl: HTMLButtonElement;
+
+    let cardUsernameEl: HTMLSpanElement;
+    let verifiedUserIdentity
+
+    const props = {
+        domain: UserType.I,
+        userId: '',
+        fieldId: 'userId',
+    }
+
+    const newProps = {
+        domain: UserType.B,
+        userId: 'newUserId',
+        fieldId: 'newFieldId',
+    };
+
+    const newValue = 'testIdir';
+
+    beforeEach(async () => {
+        wrapper = mount(UserNameInput, {
+            props,
+            global: {
+                plugins: [PrimeVue],
+            },
+        });
+
+        usernameInputText = wrapper.find('#userIdInput');
+        usernameInputTextEl = usernameInputText.element as HTMLInputElement;
+        verifyButton = wrapper.find("[data-target-btn='verifyIdir']");
+        verifyButtonEl = verifyButton.element as HTMLButtonElement;
+    });
+
+    afterEach(() => {
+        vi.clearAllMocks();
+        wrapper.unmount();
+    });
+
+    it('Should change usernameInput value', async () => {
+        await usernameInputText.setValue(newValue);
+        expect(usernameInputTextEl.value).toBe(newValue);
+    });
+
+    it('Should receive the correct prop', async () => {
+        //default props
+        expect(wrapper.props()).toEqual(props);
+
+         await wrapper.setProps(newProps);
+         expect(wrapper.props()).toEqual(newProps);
+         expect(wrapper.props()).not.toEqual(props)
+    });
+
+    it('Should call and emit correct value' , async () => {
+        await usernameInputText.setValue(newValue);
+        emitChange = wrapper.emitted('change');
+        expect(wrapper.emitted('change')).toBeTruthy();
+        // test the given parameters when emitChange has been called
+        // i.e. emitChange = [ [ 'B' ] ]
+        expect(emitChange![0][0]).toEqual(newValue);
+
+        emitSetVerifyResult = wrapper.emitted('setVerifyResult')
+        await wrapper.setProps(newProps);
```

Comment:
and maybe no need to set the props again?

Reply:
without the props I would need to change the expect to look for the emit when domain is "I"

Reply:
um.... I think the wrapper by default is rendered with props with domain "I"?

Reply:
yes, it is domain "I" by default

Reply:
oh! I see, I'll fix it!